custom-plugin-decorators/tag-sorting/README.md

```diff
@@ -0,0 +1,95 @@
+# Decorator to sort tags in an API description
+
+Authors:
+- [`@lornajane`](https://github.com/lornajane), Lorna Mitchell (Redocly)
+ 
+## What this does and why
+
+Redocly CLI has the [`tags-alphabetical'](https://redocly.com/docs/cli/rules/tags-alphabetical/) rule to error if the `tags` section isn't in alphabetical order by `name`.
+
+This decorator provides functionality to _put_ the tags in order by name, in order to make any API description compatible with that rule.
+
+## Code
+
+Here's the main plugin entrypoint, it's in `tag-sorting.js`:
+
+```javascript
+
+const SortTagsAlphabetically = require('./decorator-alpha.js');
+
+module.exports = {
+  id: 'tag-sorting',
+  decorators: {
+    oas3: {
+      'alphabetical': SortTagsAlphabetically,
+    }
+  }
+	
+}
+```
+
+The code here sets the plugin name to `tag-sorting` and adds a decorator for OpenAPI 3 called `alphabetical`.
+
+The functionality for the alphabetical sorting is in `decorator-alpha.js`:
+
+```javascript
+
+module.exports = SortTagsAlphabetically;
+
+function SortTagsAlphabetically() {
+  console.log("re-ordering tags: alphabetical");
+  return {
+    TagList: {
+      leave(target) {
+        target.sort((a,b) => {
+          if (a.name < b.name) {
+            return -1;
+          }
+        });
+      }
+    }
+  }
+}
+```
+
+It operates on the `TagList` element, since it needs to shuffle the child entries inside it. A simple comparison of the name field is used to sort the items.
+
+## Examples
```

Comment:
Thanks for showing how to use it! 

Q -- what if we had an iterative OpenAPI spec to show the various tools in use as they're added? Sort of similar to the e2e tests in the Redocly CLI repo.

Reply:
We could also maintain the examples separately (on a per-example basis), and then use the join command to run the linter and apply the decorators? 
Not a blocker -- just a nice-to-have in the future.

Reply:
I think the template is already a pretty big hurdle for outside contributors (our main audience).  This project should be self-sufficient without needing a lot of maintenance since we don't have DevRel resource. I'm not at all in favour of any of these suggestions, and I hope that doesn't come across too harshly.